Add main file Emalgo main file, add .Rproj file, refactor and rename functio…

MonkeyDflo opened the pull request:
### What my PR is changing ?
Add main file Emalgo.
Add .Rproj file.
Add previous code and comments I wrote.
Refactor and rename function splitDatasetIntoCatAndConti.  

### PR changelog (list descriptions of commits)
Just one commit.

### Notes
Need to clean code !

Commit:
Add main file Emalgo..., add .Rproj file, refactor and rename function splitDatasetIntoCatAndConti, need to clean code !

MonkeyDflo commented on EMalgo_continous_categorical_data.R:
```diff
@@ -0,0 +1,124 @@
+# packages ####
+library(bayess)
+library(mvtnorm)
+
+
+# FLO function TransformCategoricalVectorToBinaryMatrice ####
+TransformCategoricalVectorToBinaryMatrice <- function (categoricalVector) {
```
Add Description to the function.
Try to find shorter name.

MARCTELLY replied:
Oui comme je disais, le nommage doit rester accèssible étant donnée que c'est un package qui à pour objectif d'être utiliser par d'autre et non une boite noire propre à nous donc il faudrait trouver un compromis entre lisibilité des fonction et limite de longueur des noms des fonctions

MonkeyDflo commented on EMalgo_continous_categorical_data.R:
```diff
@@ -0,0 +1,124 @@
+TransformCategoricalVectorToBinaryMatrice <- function (categoricalVector) {
+  vectorUniqueValues <- unique(categoricalVector)
+  binaryMatrice <- matrix(0, nrow=length(categoricalVector), ncol=length(vectorUniqueValues))
+  for (i in 1:length(categoricalVector)) {
+    binaryMatrice[i, which(vectorUniqueValues == categoricalVector[i])] <- 1
+  }
```
Maybe we can do this with a lapply ?

MARCTELLY replied:
Dejà avant d'optimiser vec lapply il faudrait que cette function fasse tout ce dont on veut d'elle

MonkeyDflo replied:
c'est à dire ? tu peux être plus explicite @MARCTELLY

MonkeyDflo commented on EMalgo_continous_categorical_data.R:
```diff
@@ -0,0 +1,124 @@
+  return(binaryMatrice)
+}
+
+# FLO function BinarizeCategorialMatrice #####
+BinarizeCategoricalMatrice = function(CategoricalMatrice) {
```
Description

MonkeyDflo commented on EMalgo_continous_categorical_data.R:
```diff
@@ -0,0 +1,124 @@
+  return(list(continuous = continuous, categorical = categorical))
+}
+
+ds_test = iris
+res = SplitDataSetIntoCateAndContiMatrices(ds_test)
+x = cbind(res$CategoricalMatrix, res$ContinuousMatrix)
+x = getP(x)
+res = cutDatasetIntoCatAndCat(ds_test)
+# TransformCategoricalVectorToBinaryMatrice(res$CategoricalMatrix)
+# xBinary = BinarizeCategoricalMatrice(res$CategoricalMatrix)
+# print(xBinary)
```
Clean, faire un fichier de test à côté

MonkeyDflo reviewed (commented):
@MARCTELLY est-ce que tu as des remarques ?

Commit:
Refactor README add useful descriptions and comments on how to set up the project.

MARCTELLY commented on EMalgo_continous_categorical_data.R:
```diff
@@ -0,0 +1,124 @@
+  }
+}
+
+# FLO MBL-1 function CutDataSetIntoCategoricalandContinous ####
```
Il faudrait que t'ajoute à la function la condition class(data) == "character" au cas ou les dnnées ne sont pas encore encodé en catégorielle

MARCTELLY commented on EMalgo_continous_categorical_data.R:
```diff
@@ -0,0 +1,124 @@
+
+# FLO MBL-1 function CutDataSetIntoCategoricalandContinous ####
+splitDatasetIntoCatAndConti <- function(data) {
+  categorical <- data[lapply(data, class) == "factor"]
```
C'est sur cette ligne

MARCTELLY reviewed (changes requested):
Je t'ai laissé les commentaires essaie de voire comment améliorer

Commit:
-:sparkles: Add organize_dataset file to split and make it more readable.
-:sparkles: Add documentations for each function with roxygen2
-:recycle: Refactor SplitDatasetIntoCatAndConti, now can handle factor types and char type.
-:recycle: Refactor EMalgo to get functions from organize_dataset

Commit:
- :sparkles: Add many tests for organize_dataset functions in a new test_organize_dataset R file
- :recycle: Refactor previous tests

MonkeyDflo commented:
### MBL-1 Organize dataset
Important PR since we reorganize code with specific functions in specific files like organize_dataset.R file, same approach for tests files.

MonkeyDflo commented:
Accidentaly closed
